biodatasets/pubtator_central/pubtator_central.py

```diff
@@ -0,0 +1,317 @@
+exploring and retrieving bioconcept annotations in full text biomedical articles. PTC provides
+automated annotations from state-of-the-art text mining systems for genes/proteins, genetic
+variants, diseases, chemicals, species and cell lines, all available for immediate download. PTC
+annotates PubMed (30 million abstracts), the PMC Open Access Subset and the Author Manuscript
+Collection (3 million full text articles). Updated entity identification methods and a
+disambiguation module based on cutting-edge deep learning techniques provide increased accuracy.
+"""
+
+_HOMEPAGE = "https://www.ncbi.nlm.nih.gov/research/pubtator/"
+
+# TODO: Add the licence for the dataset here (if possible)
+# Note that this doesn't have to be a common open source license.
+# Some datasets have custom licenses. In this case, simply put the full license terms
+# into `_LICENSE`
+_LICENSE = ""
+
+_URLS = {
+    "api": "https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/pubtator",
+    "source": "https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/bioconcepts2pubtatorcentral.offset.gz",
+    "bigbio_kb": "https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/bioconcepts2pubtatorcentral.offset.gz",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.NAMED_ENTITY_DISAMBIGUATION]
+
+_SOURCE_VERSION = "2022.01.08"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class PubtatorCentralDataset(datasets.GeneratorBasedBuilder):
+    """PubTator Central"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="pubtator_central_api",
+            version=SOURCE_VERSION,
+            description="PubTator Central API schema",
+            schema="api",
+            subset_id="pubtator_central",
+        ),
+        BigBioConfig(
+            name="pubtator_central_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central source schema",
+            schema="source",
+            subset_id="pubtator_central",
+        ),
+        BigBioConfig(
+            name="pubtator_central_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "pubtator_central_source"
+
+    # Maps the entity types in PubTator to the name of the database they are grounded to
+    PUBTATOR_TYPE_TO_UNIFIED_DB_NAME = {
+        "Gene": "ncbi_gene",
+        "Chemical": "mesh",
+        "Species": "ncbi_taxon",
+        "Disease": "mesh",
+        "Mutation": "ncbi_dbsnp",
+        "CellLine": "cellosaurus",
+        "SNP": "ncbi_dbsnp",
+        # TODO: I don't know what these are being grounded to. It's not documented anywhere AFAICT.
+        "ProteinMutation": None,
+        "DNAMutation": None,
+        "DomainMotif": None,
+    }
```

Comment:
do these db names come from the data itself? 
is it worthwhile to put urls in this module somewhere (e.g. https://www.ncbi.nlm.nih.gov/gene/) ?

do you get anything interesting when googling some of the IDs that come back for the 3 that you have `None` for above? sometimes you get lucky ... e.g. googling "rs113488022" will lead you to ncbi_dbsnp

Reply:
I don't believe so, but they do come from PubTator, see an example below:

```
```

I agree that URLs would be better. Is there a certain key/field of the `bigbio_kb` schema that I could use? Or should we consider adding a key, e.g. `db_url`

Reply:
good question ... I was thinking more of a module level constant for the URLs ... hopefully at the end of the hackathon we can produce something like a mapping between every db_name and a URL in some utils module.